modules/tournament2/timetable.php

```diff
@@ -37,32 +39,31 @@
         $maxtime = $endtime;
     }
 }
-$db->free_result($tournaments);
 
 if ($maxtime > $mintime + 60 * 60 * 24 * 4) {
     $maxtime = $mintime + 60 * 60 * 24 * 4;
 }
 
 $head = "<td><b>".t('Turnier')."</b></td>";
 for ($z = $mintime; $z <= $maxtime; $z+= (60 * 60 * 2)) {
-    $head .= "<td colspan = 4>". $func->unixstamp2date($z, "time")."</td>";
+    $head .= "<td colspan = 4>". $func->unixstamp2date($z, "shortdaytime")."</td>";
 }
 $smarty->assign('head', $head);
 
 // Generate Table-foot
 $rows = "";
-$tournaments = $db->qry("
-  SELECT
+$tournaments = $database->queryWithFullResult(
+    "SELECT
     *,
```

Comment:
Same here about listing the explicit fields.

Reply:
This was actually solved by removing the query alltogether, as it is a 1:1 duplicate of the one above and the whole resultset has already been fetched into `$tournaments` and thus can just be iterated through again